[Bug] Can't parse past contests correctly.

### leetgo debug

root@xxx ~/leetcode (master) [2]# lc debug
Leetgo version info  :
```
dev
commit: HEAD
built at: unknown
goos: linux
goarch: amd64
```
Home dir             : /root/.config/leetgo
Project root         : /root/leetcode
Working dir          : /root/leetcode
Project config file  : /root/leetcode/leetgo.yaml
Project configuration:
```yaml
author: none
language: en
code:
  lang: python3
  cpp:
    cxx: clang++
    cxxflags: -O2 -std=c++23
  python3:
    blocks:
    - name: beforeMarker
      template: |

        from bisect import bisect_left, bisect_right
        from collections import Counter, defaultdict, deque
        from functools import cache, cmp_to_key, lru_cache, reduce
        from heapq import heapify, heappop, heappush, heappushpop, heapreplace
        from itertools import accumulate, chain, count, pairwise, zip_longest
        from math import ceil, comb, floor, gcd, inf, isqrt, lcm, log2, perm, sqrt
        from operator import xor
        from pprint import pprint
        from string import ascii_lowercase
        from typing import List, Optional
  go:
    modifiers:
      - name: removeUselessComments
      - name: changeReceiverName
      # - name: addNamedReturn
      # - name: addMod       
  # rust:
  # blocks:
  #   - name: beforeCode
  #     template: |
  #       use std::cmp::Reverse;
  #       use std::collections::*;
  #       use std::mem::swap;
leetcode:
  site: https://leetcode.com
  credentials:
    from: cookies
contest:
  open_in_browser: false
editor:
  use: custom
  command: "code"
  args: "{{.CodeFile}} {{.DescriptionFile}}"


```
Full configuration   :
```yaml
author: none
language: en
code:
  lang: python3
  filename_template: '{{ .Id | padWithZero 4 }}{{ if .SlugIsMeaningful }}.{{ .Slug }}{{ end }}'
  separate_description_file: true
  modifiers:
    - name: removeUselessComments
  go:
    out_dir: go
    modifiers:
      - name: removeUselessComments
      - name: changeReceiverName
  python3:
    out_dir: python
    blocks:
      - name: beforeMarker
        template: |2
          from bisect import bisect_left, bisect_right
          from collections import Counter, defaultdict, deque
          from functools import cache, cmp_to_key, lru_cache, reduce
          from heapq import heapify, heappop, heappush, heappushpop, heapreplace
          from itertools import accumulate, chain, count, pairwise, zip_longest
          from math import ceil, comb, floor, gcd, inf, isqrt, lcm, log2, perm, sqrt
          from operator import xor
          from pprint import pprint
          from string import ascii_lowercase
          from typing import List, Optional
    executable: python3
  cpp:
    out_dir: cpp
    cxx: clang++
    cxxflags: -O2 -std=c++23
  rust:
    out_dir: rust
  java:
    out_dir: java
leetcode:
  site: https://leetcode.com
  credentials:
    from:
      - cookies
    browsers: []
contest:
  out_dir: contest
  filename_template: '{{ .ContestShortSlug }}/{{ .Id }}{{ if .SlugIsMeaningful }}.{{ .Slug }}{{ end }}'
  open_in_browser: false
editor:
  use: custom
  command: code
  args: '{{.CodeFile}} {{.DescriptionFile}}'
```



### Debug log

```shell

```

### Description

```fish
root@xxx ~/leetcode (master) [2]# lc contest 
? Select a contest:  [Use arrows to move, type to filter]
>   Biweekly Contest 159 at 2025/06/21 22:30:00
    Weekly Contest 455 at 2025/06/22 10:30:00
× interrupt
root@xxx ~/leetcode (master) [1]# lc contest w454
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/PuerkitoBio/goquery.(*Selection).Get(...)
        /root/go/pkg/mod/github.com/!puerkito!bio/goquery@v1.10.2/array.go:77
github.com/j178/leetgo/leetcode.parseContestHtml({0xc0007da000, 0x29256, 0x2a000}, {0xc0007945b3, 0x1e}, {0x11229e2, 0x14})
        /root/leetgo/leetcode/client.go:703 +0x12b2
github.com/j178/leetgo/leetcode.(*usClient).GetContestQuestionData(0xc0004ca440, {0xc00017d2c0?, 0x41b2a5?}, {0xc0007945b3, 0x1e})
        /root/leetgo/leetcode/client_us.go:213 +0x3f2
github.com/j178/leetgo/leetcode.(*QuestionData).Fulfill(0xc0005ea288)
        /root/leetgo/leetcode/question.go:361 +0x51
github.com/j178/leetgo/lang.generate(0xc0005ea288)
        /root/leetgo/lang/gen.go:49 +0x71
github.com/j178/leetgo/lang.GenerateContest(0xc000262000)
        /root/leetgo/lang/gen.go:138 +0xdf
github.com/j178/leetgo/cmd.init.func3(0x1dfcac0, {0xc0005972b0, 0x1, 0x10f8cd6?})
        /root/leetgo/cmd/contest.go:210 +0x6e5
github.com/spf13/cobra.(*Command).execute(0x1dfcac0, {0xc000597270, 0x1, 0x1})
        /root/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1015 +0xaaa
github.com/spf13/cobra.(*Command).ExecuteC(0x1dfee80)
        /root/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1148 +0x46f
github.com/spf13/cobra.(*Command).Execute(...)
        /root/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1071
github.com/j178/leetgo/cmd.Execute()
        /root/leetgo/cmd/root.go:52 +0x1e
main.main()
        /root/leetgo/main.go:6 +0xf
root@xxx~/leetcode (master) [2]# 
```

It seems that LeetCode has changed the UI of contest, so the HTML parser can't correctly find the questions.

--- a/leetcode/client.go
+++ b/leetcode/client.go
@@ -699,101 +699,35 @@ func parseContestHtml(html []byte, questionSlug string, site config.LeetcodeSite
 	if err != nil {
 		return nil, err
 	}
-	difficulty := strings.TrimSpace(doc.Find("span.pull-right.label.round").Text())
-	frontendId := strings.TrimSuffix(doc.Find("div.question-title h3").Get(0).FirstChild.Data, ". ")
-	defaultContent, err := doc.Find("div.question-content.default-content").Html()
-	if err != nil {
-		return nil, err
+
+	script := doc.Find("script#__NEXT_DATA__")
+	if script.Length() == 0 {
+		return nil, errors.New("get contest question data: empty script")
 	}
-	sourceContent, err := doc.Find("div.question-content.source-content").Html()
-	if err != nil {
-		return nil, err
+	jsonText := script.Text()
+	result := gjson.Get(jsonText, "props.pageProps.dehydratedState.queries.#.state.data.contestQuestion.question")
+	if !result.Exists() {
+		return nil, errors.New("contest question data not found in __NEXT_DATA__")
 	}
 
-	var (
-		questionId         string
-		scriptText         string
-		codeDefinitionText string
-		metaDataText       string
-		title              string
-		sourceTitle        string
-		exampleTestcases   string
-		sampleTestcase     string
-		categoryTitle      string
-	)
-	for _, node := range doc.Find("script").Nodes {
-		if node.FirstChild != nil && strings.Contains(node.FirstChild.Data, "var pageData") {
-			scriptText = node.FirstChild.Data
+	var questionRaw []byte
+	for _, q := range result.Array() {
+		if q.Exists() && len(q.Map()) > 0 {
+			questionRaw = []byte(q.Raw)
 			break
 		}
 	}
-	if scriptText == "" {
-		return nil, errors.New("question data not found")
-	}
-	scriptLines := strings.Split(scriptText, "\n")
-	for _, line := range scriptLines {
-		switch {
-		case strings.HasPrefix(line, `    questionId: '`):
-			questionId = line[len(`    questionId: '`) : len(line)-2]
-		case strings.HasPrefix(line, `    questionTitle: '`):
-			title = line[len(`    questionTitle: '`) : len(line)-2]
-		case strings.HasPrefix(line, `    questionSourceTitle: '`):
-			sourceTitle = line[len(`    questionSourceTitle: '`) : len(line)-2]
-		case strings.HasPrefix(line, `    questionExampleTestcases: '`):
-			exampleTestcases = line[len(`    questionExampleTestcases: '`) : len(line)-2]
-			exampleTestcases = utils.DecodeRawUnicodeEscape(exampleTestcases)
-		case strings.HasPrefix(line, `    sampleTestCase: '`):
-			sampleTestcase = line[len(`    sampleTestCase: '`) : len(line)-2]
-			sampleTestcase = utils.DecodeRawUnicodeEscape(sampleTestcase)
-		case strings.HasPrefix(line, `    codeDefinition: `):
-			codeDefinitionText = line[len(`    codeDefinition: `):len(line)-len(",],")] + "]"
-			codeDefinitionText = strings.ReplaceAll(codeDefinitionText, "'", `"`)
-		case strings.HasPrefix(line, `    metaData: `):
-			metaDataText = line[len(`    metaData: JSON.parse('`) : len(line)-len(`' || '{}'),`)]
-			metaDataText = utils.DecodeRawUnicodeEscape(metaDataText)
-		case strings.HasPrefix(line, `    categoryTitle: '`):
-			categoryTitle = line[len(`    categoryTitle: '`) : len(line)-2]
-		}
+	if len(questionRaw) == 0 {
+		return nil, errors.New("contest question data not found in __NEXT_DATA__")
 	}
 
-	if site == config.LeetCodeUS {
-		sourceContent = defaultContent
-		defaultContent = ""
-		sourceTitle = title
-		title = ""
-	}
-	q := &QuestionData{
-		QuestionId:         questionId,
-		QuestionFrontendId: frontendId,
-		TitleSlug:          questionSlug,
-		Difficulty:         difficulty,
-		Content:            sourceContent,
-		TranslatedContent:  defaultContent,
-		Title:              sourceTitle,
-		TranslatedTitle:    title,
-		ExampleTestcases:   exampleTestcases,
-		SampleTestCase:     sampleTestcase,
-		CategoryTitle:      CategoryTitle(categoryTitle),
-	}
-	err = json.Unmarshal([]byte(metaDataText), &q.MetaData)
+	var qd QuestionData
+	err = json.Unmarshal(questionRaw, &qd)
 	if err != nil {
 		return nil, err
 	}
-	var codeDefs []map[string]string
-	err = json.Unmarshal([]byte(codeDefinitionText), &codeDefs)
-	if err != nil {
-		return nil, err
-	}
-	for _, codeDef := range codeDefs {
-		q.CodeSnippets = append(
-			q.CodeSnippets, CodeSnippet{
-				LangSlug: codeDef["value"],
-				Lang:     codeDef["text"],
-				Code:     codeDef["defaultCode"],
-			},
-		)
-	}
-	return q, nil
+	qd.TitleSlug = questionSlug
+	return &qd, nil
 }
 
 // 每次 "运行代码" 会产生两个 submission, 一个是运行我们的代码，一个是运行标程。

